### PR TITLE
Remove size limit on VerifiedBy column and truncate exports to 200 characters

### PR DESCRIPTION
**Pull request by dtgreiner**
## _Merging this PR_
- use the squash-merge strategy for PRs targeting a release-X branch
- use a merge-commit or rebase strategy for PRs targeting the stable branch

## Description
[//]: # (Summarize changes and include links related issue. List any new dependencies)
Remove the size limit on CurrentLivingSituation.VerifiedBy and truncate the field to 200 characters in exports

## Type of change
[//]: # (e.g., Bug fix, New feature, Code clean-up, Dependency update)
Bug fix

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [X] I have performed a self-review of my code
- [X] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [ ] If adding a new endpoint / exposing data in a new way, I have:
  - [ ] ensured the API can't leak data from other data sources
  - [ ] ensured this does not introduce N+1s
  - [ ] ensured permissions and visibility checks are performed in the right places
- [ ] I have updated the documentation (or not applicable)
- [ ] I have added spec tests (or not applicable)
- [ ] I have provided testing instructions in this PR or the related issue (or not applicable)

### CODE REVIEW

File: db/warehouse/migrate/20241030133448_remove_current_living_situation_verified_by_limit.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class RemoveCurrentLivingSituationVerifiedByLimit < ActiveRecord::Migration[7.0]
+  def change
+    # view 'bi_CurrentLivingSituation' depends on column 'VerifiedBy'. Need to remove it before making this change and recreate it afterwards.
+    safety_assured do
+      drop_view 'bi_CurrentLivingSituation', revert_to_version: 1
```
Comment (eanders) [inline db/warehouse/migrate/20241030133448_remove_current_living_situation_verified_by_limit.rb]:
```suggestion
    BI::ViewMaintainer.new.safe_drop_view(view_name(GrdaWarehouse::Hud::CurrentLivingSituation))
```

File: db/warehouse/migrate/20241030133448_remove_current_living_situation_verified_by_limit.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      reversible do |dir|
+        dir.up do
+          change_column :CurrentLivingSituation, :VerifiedBy, :string, :limit => nil
+        end
+        dir.down do
+          change_column :CurrentLivingSituation, :VerifiedBy, :string, :limit => 100
+        end
+      end
+      create_view 'bi_CurrentLivingSituation'
```
Comment (eanders) [inline db/warehouse/migrate/20241030133448_remove_current_living_situation_verified_by_limit.rb]:
```suggestion
      BI::ViewMaintainer.new.safe_create_view(view_name(GrdaWarehouse::Hud::CurrentLivingSituation))
```

**Comment by dtgreiner** [comment]
@eanders This should be ready for another review. I ended up having to split up the migration into multiple migrations due to running into lock issues when attempting to run it all in a single migration, but it is using the existing view maintainer now.

**Review by eanders — APPROVED** [review]
Looks good thanks!  I'm going to tag @gigxz to make sure the HMIS side is covered sufficiently.

**Review by gigxz — COMMENTED** [review]
Looks like there is a test around the truncation that would need to be updated https://github.com/greenriver/hmis-warehouse/actions/runs/11614155280/job/32341650418

Otherwise looks good to me. Thanks!

**Comment by dtgreiner** [comment]
@gigxz @eanders We are no longer truncating the VerifiedBy field at the point the unit test is running but instead doing truncation on the export report. I've removed this test, but happy to revisit if we think we need this done in another way.

**Comment by gigxz** [comment]
The truncation remains in file `drivers/hmis/app/models/hmis/hud/processors/current_living_situation_processor.rb` so it is still truncating at point of collection in HMIS. But that's fine. The field is only used for export anyway (as far as HMIS is concerned). Removing the test is ok.

**Review by gigxz — APPROVED** [review]

**Pull request merged**